### ISSUE TITLE
Allow using sandman as a blueprint rather than an app

### ISSUE THREAD

**Issue by pib** [jeffknupp/sandman#6]
It would be useful to have sandman optionally usable as a [Blueprint](http://flask.pocoo.org/docs/blueprints/).

As it is right now, sandman works well if _all_ you want is a REST API, but it's difficult and/or impossible to add a REST API to an existing app without hacking your app on top of the app sandman automatically builds.

It should be possible to just do something like:

```
from flask import Flask
from sandman import blueprint as sandman_blueprint

app = Flask(__name__)
app.register_blueprint(sandman_blueprint, url_prefix='/api')
app.register_resource((Model, OtherModel))
```

Or I suppose instead of adding a method to the app object, there could just be an optional app argument to sandman.register.

Basically, I'm saying sandman should be able to function without using any global shared objects. That goes for the db global as well. One of the nice things about Flask is that it does away with all the "magic globals" that so many other frameworks have, so it'd be nice to preserve that with sandman which otherwise seems like a very nice library!

**Comment by jeffknupp** [comment]
Actually, the strong coupling with Flask is going to be broken in the future. The idea is to support a number of Python web frameworks via an abstracted "API service engine" (sort of like a SQLAlchemy for web frameworks to be used by Sandman).

Basically, I understand why those who have existing Flask apps might find it more useful as a blueprint,  but I think the library would be more useful in general without being tightly coupled to _any_ framework.
